tests/utils/mock_microgrid.py

```diff
@@ -0,0 +1,207 @@
+from google.protobuf.empty_pb2 import Empty  # pylint: disable=no-name-in-module
+
+from frequenz.sdk.microgrid import Microgrid
+from frequenz.sdk.microgrid._graph import ComponentGraph, _MicrogridComponentGraph
+from frequenz.sdk.microgrid.client import Connection
+from frequenz.sdk.microgrid.component import (
+    BatteryData,
+    Component,
+    ComponentCategory,
+    InverterData,
+)
+
+
+class MockMicrogridClient:
+    """Class that mocks MicrogridClient behavior."""
+
+    def __init__(self, components: Set[Component], connections: Set[Connection]):
+        """Create mock microgrid with given components and connections.
+
+        This simulates microgrid.
+        Every call to `battery_data` and `inverter_data` is mocked to return
+        receiver owned by this class.
+        User can send data to the receiver using `self.send(..)` method.
+        Messages will be send based on component id.
+
+        Args:
+            components: List of the microgrid components
+            connections: List of the microgrid connections
+        """
+        self._component_graph = _MicrogridComponentGraph(components, connections)
+
+        bat_channels = self._create_battery_channels()
+        inv_channels = self._create_inverter_channels()
+
+        mock_api = self._create_mock_api(components, bat_channels, inv_channels)
+        kwargs: Dict[str, Any] = {
+            "api_client": mock_api,
+            "component_graph": self._component_graph,
+        }
+
+        self._mock_microgrid = MagicMock(spec=Microgrid, **kwargs)
+        self._battery_data_senders = {
+            id: channel.new_sender() for id, channel in bat_channels.items()
+        }
+        self._inverter_data_senders = {
+            id: channel.new_sender() for id, channel in inv_channels.items()
+        }
+
+    @property
+    def mock_microgrid(self) -> Microgrid:
+        """Return mock microgrid.
+
+        This is needed to patch existing microgrid.get() method.
+
+        Returns:
+            Mock microgrid.
+        """
+        return self._mock_microgrid
+
+    @property
+    def component_graph(self) -> ComponentGraph:
+        """Return microgrid component graph.
+
+        Component graph is not mocked.
+
+        Returns:
+            Mock microgrid.
+        """
+        return self._component_graph
+
+    async def send(self, data: Union[BatteryData, InverterData]) -> bool:
+        """Send component data using channel.
+
+        This simulates component sending data. Right now only battery and inverter
+        are supported. More components categories can be added if needed.
+
+        Args:
+            data: Data to be send
+
+        Returns:
+            Whether data was send, or not.
+        """
+        cid = data.component_id
+        if isinstance(data, BatteryData):
+            return await self._battery_data_senders[cid].send(data)
+
+        return await self._inverter_data_senders[cid].send(data)
```

Comment:
could you change to:
``` python
if BatteryData:
    send
elif InverterData:
    send
else:
    raise
```

and update function signature to `data: ComponentData`?  Then we won't have a large param type as we add more types.

Reply:
Done